carla/recourse_methods/autoencoder/losses/losses.py

```diff
@@ -1,6 +1,64 @@
 import numpy as np
+import torch
+import torch.distributions as dists
 from keras import backend as K
+from torch import nn
 
 
 def binary_crossentropy(y_true: np.ndarray, y_pred: np.ndarray) -> np.ndarray:
     return K.sum(K.binary_crossentropy(y_true, y_pred), axis=-1)
+
+
+def csvae_loss(csvae, x_train, y_train):
```

Comment:
I can understand from the code how this works, however adding some comments might make it a bit easier. E.g. that the x_mu, x_logvar define a normal distribution. It's low priority though.

Reply:
maybe also refer to an equation in a paper or something? I'm not sure where exactly this specific loss comes from